cypress/cypress/views/submariner/actions/submariner_actions.js

```diff
@@ -6,24 +6,17 @@
 /// <reference types="cypress" />
 
 import { clusterSetMethods } from '../../../views/clusterset/clusterset'
+import { clustersPages } from '../../../views/clusters/managedCluster'
 
 export const submarinerClusterSetMethods = {
 
-    // The function verifies that submariner deployment exists.
+    // The function verifies that submariner cluster exists.
     submarinerClusterSetShouldExist: (clusterSetName) => {
-        cy.log("verify that a cluster set with deployment of submariner is exists")
-        clusterSetMethods.clusterSetShouldExist(clusterSetName)
+        cy.log("verify that a cluster set named " + clusterSetMethods + " exists")
+        clustersPages.goToClusterSet()
+        cy.get('.pf-c-text-input-group__text-input').type(clusterSetName)
 
-        cy.get('[data-label=Name]').eq(1).then(($clusterset) => {
-            if ($clusterset.text().includes(clusterSetName)) {
-                cy.log("A cluster set named " + clusterSetName + " was found")
-                cy.get('[data-label=Name]').contains(clusterSetName).click()
-                cy.get('.pf-c-nav__link').contains('Submariner add-ons').click()
-            }
-            else{
-                cy.log("A cluster set named " + clusterSetName + " was not found. can't continue with the test")
-            }
-        })
+        return cy.get('[data-label=Name]')
```

Comment:
Not just the definition, but the whole logic. This is because for each test a different operation is performed if the cluster set is found or not. For example, in the deployment test, if the cluster set is not found, it is created. But in the validation test, if there is no such cluster set, then the test is not performed at all.
Originally I wanted the function to return true or false if the cluster set was found or not. But this is not possible in cypress when an asynchronous function is used, so I had to return the value itself `return cy.get('[data-label=Name]')` and check outside the function what its value was.

Reply:
@NogaTal98 

Ok, but how is it relates to the log message?
I mean, if next steps of the test depends on the test flow, it doesn't really related to the log message.

All you have to do, is to print the log output if the clsuterset exists or not right before you return.
No?

Reply:
@MaxBab 
But I dont know if it is exists or not. That is what I explained, that the existent check is outside of the method.

Reply:
@MaxBab 
Not exactly, because for that I need to call the object twice and perform the test that in any case should be performed outside the function, inside the function one more time. 
But I will do it.
I'm deploying an environment to test the changes and then I'll push them.

Reply:
I got your point.
No need to perform steps twice.

I'm fine with that way too.